agent/core/src/main/java/org/jolokia/detector/LightstreamerDetector.java

```diff
@@ -0,0 +1,68 @@
+package org.jolokia.detector;
+
+import java.lang.instrument.Instrumentation;
+import org.jolokia.backend.executor.MBeanServerExecutor;
+
+public class LightstreamerDetector extends AbstractServerDetector {
+
+
+    /** {@inheritDoc}
+     * @param pMBeanServerExecutor*/
+    public ServerHandle detect(MBeanServerExecutor pMBeanServerExecutor) {
+        String serverVersion = getSingleStringAttribute(pMBeanServerExecutor, "com.lightstreamer:type=Server", "LSVersion");
+        if (serverVersion != null) {
+            return new ServerHandle("LightStreamer", "LightStreamer", serverVersion, null);
+        }
+        return null;
+    }
+
+    @Override
+    public void jvmAgentStartup(Instrumentation instrumentation) {
+        if (isLightStreamer(instrumentation)) {
+            awaitLightstreamerMBeans(instrumentation);
+        }
+    }
+    
+    protected boolean isLightStreamer(Instrumentation instrumentation) {
+        for (String name : System.getProperties().stringPropertyNames()) {
+            if (name.contains("com.lightstreamer")) {
+                return true;
+            }
+        }
+        for (Class loadedClass : instrumentation.getAllLoadedClasses()) {
```

Comment:
Isn't this a bit heavy weight to iterate over all loaded classes ? This will slow down every startup even when not running on lightstream.